### PR TITLE
add docs for optimizers

### PR DESCRIPTION
**Pull request by ferrine**

### CODE REVIEW

File: pymc3/variational/updates.py [hunk 1]
```diff
@@ -209,6 +226,7 @@ def sgd(loss_or_grads, params, learning_rate=1e-3):
 
 
 class Sgd(Optimizer):
+    __doc__ = _make_doc(sgd.__doc__)
```
Comment (twiecki) [inline pymc3/variational/updates.py:229]: Another option would be to move this into the base class `__new__` and do `__doc__ = _make_doc(self._opt.__doc__)`.

Comment (ferrine) [reply pymc3/variational/updates.py:229]: great point

Comment (ferrine) [reply pymc3/variational/updates.py:229]: That does not work, metaclass magic is needed

Comment (ferrine) [reply pymc3/variational/updates.py:229]: I don't think it is reasonable here

**Comment by twiecki** [comment]
Agreed.

On May 5, 2017 12:26 AM, "Maxim Kochurov" <notifications@github.com> wrote:

Merged #2119 <https://github.com/pymc-devs/pymc3/pull/2119>.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/pymc-devs/pymc3/pull/2119#event-1069780467>, or mute
the thread

**Pull request merged**